arrow-libs/optics/arrow-optics/src/main/kotlin/arrow/optics/std/tuple.kt

```diff
@@ -1,239 +1,264 @@
  */
-fun <A, B, C, R> Tuple3.Companion.pSecond(): PLens<Tuple3<A, B, C>, Tuple3<A, R, C>, B, R> = PLens(
-  get = { it.b },
-  set = { (a, _, c), r -> Tuple3(a, r, c) }
-)
+fun <A, B, C, R> PLens.Companion.pTripleSecond(): PLens<Triple<A, B, C>, Triple<A, R, C>, B, R> =
+  PLens(
+    get = { it.second },
+    set = { (a, _, c), r -> Triple(a, r, c) }
+  )
 
 /**
- * [Lens] to focus into the second value of a [arrow.core.Tuple3]
+ * [Lens] to focus into the second value of a [arrow.core.Triple]
  */
-fun <A, B, C> Tuple3.Companion.second(): Lens<Tuple3<A, B, C>, B> = pSecond()
+fun <A, B, C> PLens.Companion.tripleSecond(): Lens<Triple<A, B, C>, B> =
+  pTripleSecond()
 
 /**
- * [PLens] to focus into the third value of a [arrow.core.Tuple3]
+ * [PLens] to focus into the third value of a [arrow.core.Triple]
  */
-fun <A, B, C, R> Tuple3.Companion.pThird(): PLens<Tuple3<A, B, C>, Tuple3<A, B, R>, C, R> = PLens(
-  get = { it.c },
-  set = { (a, b, _), r -> Tuple3(a, b, r) }
-)
+fun <A, B, C, R> PLens.Companion.pTripleThird(): PLens<Triple<A, B, C>, Triple<A, B, R>, C, R> =
+  PLens(
+    get = { it.third },
+    set = { (a, b, _), r -> Triple(a, b, r) }
+  )
 
 /**
- * [Lens] to focus into the third value of a [arrow.core.Tuple3]
+ * [Lens] to focus into the third value of a [arrow.core.Triple]
  */
-fun <A, B, C> Tuple3.Companion.third(): Lens<Tuple3<A, B, C>, C> = pThird()
+fun <A, B, C> PLens.Companion.tripleThird(): Lens<Triple<A, B, C>, C> =
+  pTripleThird()
 
 /**
- * [PTraversal] to focus into the first, second and third value of a [arrow.core.Tuple3]
+ * [PTraversal] to focus into the first, second and third value of a [arrow.core.Triple]
  */
-fun <A, B> Tuple3.Companion.pTraversal(): PTraversal<Tuple3<A, A, A>, Tuple3<B, B, B>, A, B> = PTraversal(
-  get1 = { it.a },
-  get2 = { it.b },
-  get3 = { it.c },
-  set = { a, b, c, _ -> Tuple3(a, b, c) }
-)
+fun <A, B> PTraversal.Companion.pTriple(): PTraversal<Triple<A, A, A>, Triple<B, B, B>, A, B> =
+  PTraversal(
+    get1 = { it.first },
+    get2 = { it.second },
+    get3 = { it.third },
+    set = { a, b, c, _ -> Triple(a, b, c) }
+  )
 
 /**
- * [Traversal] to focus into the first, second and third value of a [arrow.core.Tuple3]
+ * [Traversal] to focus into the first, second and third value of a [arrow.core.Triple]
  */
-fun <A> Tuple3.Companion.traversal(): Traversal<Tuple3<A, A, A>, A> = pTraversal()
+fun <A> PTraversal.Companion.triple(): Traversal<Triple<A, A, A>, A> =
+  pTriple()
 
 /**
  * [PTraversal] to focus into the first, second, third and fourth value of a [arrow.core.Tuple4]
  */
-fun <A, B> Tuple4.Companion.pTraversal(): PTraversal<Tuple4<A, A, A, A>, Tuple4<B, B, B, B>, A, B> = PTraversal(
-  get1 = { it.a },
-  get2 = { it.b },
-  get3 = { it.c },
-  get4 = { it.d },
-  set = { a, b, c, d, _ -> Tuple4(a, b, c, d) }
-)
+fun <A, B> PTraversal.Companion.pTuple4(): PTraversal<Tuple4<A, A, A, A>, Tuple4<B, B, B, B>, A, B> =
+  PTraversal(
+    get1 = { it.a },
```

Comment:
We have deprecated the `a` field in tuples in favor of `first` in `0.12` so we should remove it in the version `0.13`, shouldn't we?

Reply:
Yes, absolutely!

I want to take such changes on in seperate PRs, in this case the one where we remove `Tuple2` and `Tuple3` from the codebase. Or when we remove `a`, `b`, `c`, ... properties from `TupleN`. That way those changes would be more grouped, and will be caught by the compiler/Ank.

Reply:
I tackled it in this PR anyway ;)